### PR TITLE
Allowing to overwrite Prithvi and new segmentation tests.

### PR DESCRIPTION
**Pull request by Joao-L-S-Almeida**
Segmentation was not being tested during fine-tuning tests.

### CODE REVIEW

**Comment by Joao-L-S-Almeida** [comment]
@CarlosGomes98 @romeokienzler The conflicts were solved.

File: terratorch/models/backbones/prithvi_vit.py [hunk 1]
```diff
@@ -139,13 +141,22 @@ def create_prithvi_vit_100(
...
         "num_frames": 1,
     }
+
+    # It is possible to overwrite default parameters using 
+    # config file
+    kwargs_ = defaultdict()
+    kwargs_.update(model_args)
+    kwargs_.update(kwargs)
+    kwargs_ = dict(kwargs_)
+
```
Comment (CarlosGomes98) [inline terratorch/models/backbones/prithvi_vit.py:151]: why is this necessary? the previous line 147 should already cause `model_args` to be overwritten if the same key exists in kwargs. 

```python
>>> a = {"b": 2}
>>> kwargs = {"b": 3, "c": 4}
>>> dict(a, **kwargs)
{'b': 3, 'c': 4}
```

File: terratorch/models/backbones/prithvi_vit.py [hunk 2]
```diff
@@ -81,7 +83,7 @@ def _create_prithvi(
     if "features_only" in kwargs:
         kwargs = {k: v for k, v in kwargs.items() if k != "features_only"}
 
-    kwargs["in_chans"] = len(model_bands)
+    kwargs["in_chans"] = _estimate_in_chans(model_bands=model_bands)
```
Comment (CarlosGomes98) [inline terratorch/models/backbones/prithvi_vit.py:86]: If the user passes `model_bands` with intervals, don't we need to "expand" this into the full form in order to use it in the `checkpoint_filter_fn`? I think this part is missing.

Then, once we have expanded, we should just be able to call `len` on that. I think we shouldn't need this `_estimate_in_chans`, the above approach should be simpler.

**Review by CarlosGomes98 — CHANGES_REQUESTED** [review]
Adding the segmentation tests seems good to me. But the other two changes I see some issues with. Happy to discuss though

File: terratorch/models/backbones/prithvi_vit.py (hunk 1 above)
Comment (Joao-L-S-Almeida) [reply terratorch/models/backbones/prithvi_vit.py:151]: That's true. I don't know why it wasn't working for my test.

File: terratorch/models/backbones/prithvi_vit.py (hunk 2 above)
Comment (Joao-L-S-Almeida) [reply terratorch/models/backbones/prithvi_vit.py]: Maybe the modification from the last commits solves this issue.

File: terratorch/datasets/generic_pixel_wise_dataset.py [hunk 3]
```diff
@@ -152,7 +152,7 @@ def __getitem__(self, index: int) -> dict[str, Any]:
         if self.expand_temporal_dimension:
             image = rearrange(image, "(channels time) h w -> channels time h w", channels=len(self.output_bands))
         image = np.moveaxis(image, 0, -1)
-
+
```
[the remark this answers is not in the document]
Comment (Joao-L-S-Almeida) [reply terratorch/datasets/generic_pixel_wise_dataset.py:155]: About it and others related issues, I'll run the codebase under ruff and send it as as separated pull request.

File: terratorch/datasets/generic_pixel_wise_dataset.py [hunk 4]
```diff
@@ -167,7 +167,7 @@ def __getitem__(self, index: int) -> dict[str, Any]:
         if self.transform:
             output = self.transform(**output)
         output["filename"] = self.image_files[index]
-
+
```
[the remark this answers is not in the document]
Comment (Joao-L-S-Almeida) [reply terratorch/datasets/generic_pixel_wise_dataset.py:170]: About it and others related issues, I'll run the codebase under ruff and send it as as separated pull request.

File: terratorch/models/backbones/utils.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+from terratorch.datasets import HLSBands
+
+def _are_sublists_of_int(item) -> (bool, bool):
```
Comment (CarlosGomes98) [inline terratorch/models/backbones/utils.py:3]: I believe this is never used. if so, can we remove it?

**Review by CarlosGomes98 — DISMISSED** [review]
After removing that function (if it really is never used) lgtm. Lets do that ruff on the codebase after then :)

**Comment by Joao-L-S-Almeida** [comment]
Done.

**Comment by Joao-L-S-Almeida** [comment]
WARNING. If this PR isn't approved (or rejected) within 7 days it will be approved as is.
